apis/logging/v1/clusterlogging_types.go

```diff
@@ -178,34 +173,67 @@ type ElasticsearchSpec struct {
 
 // This is the struct that will contain information pertinent to Log and event collection
 type CollectionSpec struct {
-	// Specification of Log Collection for the cluster
-	Logs LogCollectionSpec `json:"logs,omitempty"`
-}
 
-type LogCollectionSpec struct {
 	// The type of Log Collection to configure
 	Type LogCollectionType `json:"type"`
 
+	// The resource requirements for Collector
+	//
+	// +nullable
+	// +optional
+	Resources *v1.ResourceRequirements `json:"resources"`
+
+	// Define which Nodes the Collector Pods are scheduled on.
+	//
+	// +nullable
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	Tolerations  []v1.Toleration   `json:"tolerations,omitempty"`
+
 	// Specification of the Fluentd Log Collection component
-	FluentdSpec `json:"fluentd,omitempty"`
+	//
+	// +nullable
+	// +optional
+	FluentdSpec *FluentdSpec `json:"fluentd,omitempty"`
+
+	// Specification of the Vector Log Collection component
+	//
+	// +nullable
+	// +optional
+	VectorSpec *VectorSpec `json:"vector,omitempty"`
```

Comment:
This is an additive, backwards compatible change, OK to add this for vector support.